### PR TITLE
Implement 'cast_load_addr' tile for exprjit 

### PR DESCRIPTION
**Pull request by jstuder-gh**
Implement the 'cast_load_addr' tile for the exprjit template.
Logic mirrors that of the standard cast operation.

Also adds a template that uses the new tile and a macro for storage_spec.

Compiles and passes spectest, but definitely could use review. ping @bdw.

### CODE REVIEW

File: src/jit/x64/tiles.dasc [hunk 1]
```diff
@@ -263,8 +263,68 @@ MVM_JIT_TILE_DECL(cast) {
...
 
+/* Logic is the same as cast above except loading from memory instead of reg */
+/* See comments in cast above for more detail */
 MVM_JIT_TILE_DECL(cast_load_addr) {
-    DIE("NYI");
+    MVMint32 ofs       = tile->args[0];
+    MVMint32 load_size = tile->args[1];
+    MVMint32 to_size   = tile->args[2];
+    MVMint32 from_size = tile->args[3];
+    MVMint32 cast_mode = tile->args[4];
```
Comment (bdw) [inline src/jit/x64/tiles.dasc]: Ouch. The size of `args` is limited to 4. I guess that's not enough then.

Comment (bdw) [reply src/jit/x64/tiles.dasc]: ... although you don't strictly need both from_size and load_size, so maybe there is a fix for this

File: src/jit/core_templates.expr [hunk 2]
```diff
@@ -689,6 +689,14 @@
       (carg $2 ptr)
       (carg \$0 ptr))))
 
+(template: objprimspec
+  (if
+    (nz $1)
+    (cast
+      (^getf (^storage_spec $1) MVMStorageSpec boxed_primitive)
+      int_sz (&sizeof MVMuint16) (&QUOTE MVM_JIT_CAST_UNSIGNED))
```
Comment (bdw) [inline src/jit/core_templates.expr]: `(&QUOTE MVM_JIT_CAST_UNSIGNED)` can be written as `cast_unsigned` (the `MVM_JIT_` is an implied prefix); the `(&QUOTE)` macro expression is intended for when this isn't suitable.

File: src/jit/x64/tiles.dasc (hunk 1 above)
Comment (jstuder-gh) [reply src/jit/x64/tiles.dasc]: If we set the args array size to something higher (6 might be a good number?), would that interfere with any code expecting a certain size or alignment of the structure?

Comment (bdw) [reply src/jit/x64/tiles.dasc]: Yes, that is actually the easiest and cheapest fix.

**Comment by jstuder-gh** [comment]
@bdw, thanks for the review! I've added another commit extending the size of the args array to 6.

**Pull request merged**